dpnp/dpnp_iface_histograms.py

```diff
@@ -392,20 +426,16 @@ def bincount(x, weights=None, minlength=None):
             "supported types"
         )
 
-    x_casted = dpnp.astype(x, dtype=x_casted_dtype, copy=False)
+    x_casted = _normalize_array(x, dtype=x_casted_dtype)
 
     if weights is not None:
-        weights_casted = dpnp.astype(weights, dtype=ntype_casted, copy=False)
+        weights_casted = _normalize_array(weights, dtype=ntype_casted)
 
     n_casted = _bincount_run_native(
         x_casted, weights_casted, minlength, ntype_casted, usm_type
     )
 
-    n_usm_type = n_casted.usm_type
-    if usm_type != n_usm_type:
-        n = dpnp.asarray(n_casted, dtype=ntype, usm_type=usm_type)
-    else:
-        n = dpnp.astype(n_casted, ntype, copy=False)
+    n = _normalize_array(n_casted, dtype=ntype, usm_type=usm_type)
```

Comment:
In general, `astype` seems more lightweighted than `asarray` and so it is preferrable to use `astype`.

Reply:
`asarray` and `astype` has different set of parameters and different default behavior.
I really don't want to combine these two